clients/drcachesim/tracer/output.cpp

```diff
@@ -1220,6 +1220,9 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
                 }
             }
             if (hit_window_end) {
+                // Go to the next interval, if -trace_instr_intervals_file is set.
+                increment_window_index();
```

Comment:
There's an assumption here regarding the ordering of no-trace and trace periods. Since we increment window_idx when we hit the end of a tracing period, we assume no-tracing comes first and then tracing, and then we increment window_idx. Note this in a comment somewhere.

Reply:
Done, right after this comment.
Although, if the start of the first window is "instruction 0" the initial no-trace period of the window is also 0, which means we start with tracing right away.